src/main/java/org/opensearch/security/action/apitokens/ApiTokenRepository.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.action.apitokens;
+
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+
+public class ApiTokenRepository {
+    private final ApiTokenIndexHandler apiTokenIndexHandler;
+
+    public ApiTokenRepository(Client client, ClusterService clusterService) {
+        apiTokenIndexHandler = new ApiTokenIndexHandler(client, clusterService);
+    }
+
+    public String createApiToken(String name, List<String> clusterPermissions, List<ApiToken.IndexPermission> indexPermissions) {
+        apiTokenIndexHandler.createApiTokenIndexIfAbsent();
+        // TODO: Implement logic of creating JTI to match against during authc/z
+        // TODO: Add validation on whether user is creating a token with a subset of their permissions
+        return apiTokenIndexHandler.indexTokenMetadata(new ApiToken(name, "test-token", clusterPermissions, indexPermissions));
+    }
+
+    public void deleteApiToken(String name) throws ApiTokenException {
+        apiTokenIndexHandler.createApiTokenIndexIfAbsent();
```

Comment:
why do we delete token if Index doesn't exist?

Reply:
I have logic handling the delete of the api token if it doesn't exist. My thinking around creating the index here is that if users are trying to call on the delete of this endpoint in the first place they may be interested in this feature, so we create the index for them (even if the token doesn't exist). Again this can be changed in future/if index is created on bootstrap, but we can re visit this when we decide on the correct approach for index creation.

Reply:
We should not be creating an index. Instead we can modify the method definition to throw an exception when Index is not found.

Reply:
Same for below.